app/code/community/Paylike/Paymentmethod/Model/Paymentmethod.php

```diff
@@ -22,25 +23,44 @@ public function authorize(Varien_Object $payment, $amount)
     {
         $payment->setTransactionId($payment->getPaylikeTransactionId());
         $payment->setIsTransactionClosed(0);
-        return $this;
-
+        $paylike = new Paylike($this->getApiKey());
+        $result = $paylike->transactions->fetch($payment->getTransactionId());
+        if ($transaction = $result->transaction) {
+            if ($transaction->id != $payment->getTransactionId() || $transaction->amount != $amount * 100 || Mage::app()->getStore()->getCurrentCurrencyCode() != $transaction->currency) {
+                $errormsg = $this->__('Invalid transaction.');
+                Mage::throwException($errormsg);
+            }
+        } else {
+            $errormsg = $this->__('Invalid transaction.');
+            Mage::throwException($errormsg);
+        }
+        return $this;		
     }
 
     public function capture(Varien_Object $payment, $amount)
     {
+        $paylike = new Paylike($this->getApiKey());
         $arr = array('amount' => $amount * 100,
             'currency' => Mage::app()->getStore()->getCurrentCurrencyCode(),
-            'descriptor' => 'Capture',
+            'descriptor' => '',
         );
         if ($amount <= 0) {
-            $errormsg= $this->__('Invalid amount for authorization.');
+            $errormsg = $this->__('Invalid amount for authorization.');
             Mage::throwException($errormsg);
         }
-        $paylike = new Paylike($this->getApiKey());
         if (!$payment->getLastTransId()) {
             $payment->setLastTransId($payment->getPaylikeTransactionId());
         }
-
+        $data = $paylike->transactions->fetch($payment->getLastTransId());
+        if ($transaction = $data->transaction) {
+            if ($transaction->id != $payment->getLastTransId() || $transaction->amount != $amount * 100 || Mage::app()->getStore()->getCurrentCurrencyCode() != $transaction->currency) {
+                $errormsg = $this->__('Invalid transaction.');                
+                Mage::throwException($errormsg);
+            }
+        } else {
+            $errormsg = $this->__('Invalid transaction.');
+            Mage::throwException($errormsg);
+        }
```

Comment:
If the `authorize` method is run on every sale, then there is not reason to do this check again. It would (hopefully) always give the exact same result.